### ISSUE TITLE
Bug: [no-deprecated] doesn't report on deprecated variables used in destructuring assignment

### ISSUE THREAD

**Issue by auvred** [typescript-eslint/typescript-eslint#9968]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.5.4&fileType=.tsx&code=PQKhAIAEBMFMAcBOsDGBDALra4THCgPYB2AzhuDgLzgCMA3AFCPT3jD7LyGJY4C04QgGtmRMhQDejcAXA0cHcMUIUuPPuEEAjAK4BzRgF954SUaZA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oBNFjpEZAIb5E3dFETRoHaJHBgAviEVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
/** @deprecated */ const d = 1;

d; // reported - ok

const {
  c = d // not reported - bug
} = {};
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/no-deprecated": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

`d` should be reported in `const { c = d } = {}`

### Actual Result

`d` is not reported
